src/core/components/hv-route/index.tsx

```diff
@@ -96,8 +96,8 @@ class HvRouteInner extends PureComponent<Types.InnerRouteProps, ScreenState> {
   componentDidUpdate(prevProps: Types.InnerRouteProps) {
     if (prevProps.url !== this.props.url || this.needsLoad) {
       this.load();
+      this.needsLoad = false;
```

Comment:
is this change intentional? If so could you describe in the PR description why / how it affected the bug that's being fixed?